### ISSUE TITLE
Updated config variable handling (0.1.1) fails with vars from addons

### ISSUE THREAD

**Issue by justincampbell** [terraform-providers/terraform-provider-heroku#24]
### Terraform Version
`Terraform v0.10.8`

### Affected Resource(s)

- `heroku_app`

### Terraform Configuration Files

Start with:

```hcl
provider "heroku" {
  version = "0.1.1"
}

resource "random_id" "app_name_suffix" {
  byte_length = 4
}

resource "heroku_app" "default" {
  name   = "test-${random_id.app_name_suffix.hex}"
  region = "us"

  config_vars {
    A = "1"
    B = "2"
    C = "3"
  }
}

resource "heroku_addon" "database" {
  app  = "${heroku_app.default.name}"
  plan = "heroku-postgresql:hobby-basic"
}
```

### Steps to Reproduce

(Given config above)

1. `terraform init`
2. `terraform apply`
3. `terraform plan`
4. `terraform apply`

Step 3 shows:

```
------------------------------------------------------------------------

An execution plan has been generated and is shown below.
Resource actions are indicated with the following symbols:
  ~ update in-place

Terraform will perform the following actions:

  ~ heroku_app.default
      config_vars.0.%:            "4" => "3"
      config_vars.0.DATABASE_URL: "postgres://vslgcepyvutomi:ac56b43be294ce3da2e6f04d618038e45ac4712d08494442edac534d1ca6862d@ec2-50-16-202-213.compute-1.amazonaws.com:5432/d6tcmecqio7rgu" => ""


Plan: 0 to add, 1 to change, 0 to destroy.

------------------------------------------------------------------------
```

Step 4 shows:

```
heroku_app.default: Modifying... (ID: test-588eef73)
  config_vars.0.%:            "4" => "3"
  config_vars.0.DATABASE_URL: "postgres://vslgcepyvutomi:ac56b43be294ce3da2e6f04d618038e45ac4712d08494442edac534d1ca6862d@ec2-50-16-202-213.compute-1.amazonaws.com:5432/d6tcmecqio7rgu" => ""

Error: Error applying plan:

1 error(s) occurred:

* heroku_app.default: 1 error(s) occurred:

* heroku_app.default: Error updating config vars: Patch https://api.heroku.com/apps/test-588eef73/config-vars: Cannot overwrite attachment values DATABASE_URL.
```

**Comment by justincampbell** [comment]
This also happens if the version is not changed (just apply the config above with 0.1.1 and then do another plan/apply).

**Comment by justincampbell** [comment]
Updated the issue description to reflect my comment above.

**Comment by bernerdschaefer** [comment]
Related to https://github.com/terraform-providers/terraform-provider-heroku/pull/17?

**Comment by catsby** [comment]
Should be patched in https://github.com/terraform-providers/terraform-provider-heroku/pull/25

**Comment by justincampbell** [comment]
Fixed by #36

**Comment by justincampbell** [comment]
[v0.1.2 has been released](https://github.com/terraform-providers/terraform-provider-heroku/blob/master/CHANGELOG.md#012-january-04-2018) with a fix for this. Please test and report any issues. Thanks!
